src/AzSK.ADO/Framework/Core/SVT/ADO/ADO.AgentPool.ps1

```diff
@@ -445,6 +508,40 @@ class AgentPool: ADOSVTBase
         return $controlResult
     }
 
+    hidden [ControlResult] CheckCredInEnvironmentVariablesAutomatedFix([ControlResult] $controlResult)
+    {
+        try 
+        {
+        
+            $RawDataObjForControlFix = @();
+            $RawDataObjForControlFix = ([ControlHelper]::ControlFixBackup | where-object {$_.ResourceId -eq $this.ResourceId}).DataObject
+
+            $RawDataObjForControlFix | ForEach-Object {
+                $CurrentAgent= $_
+                if (-not $this.UndoFix)
+                {                 
+                    $body =  $CurrentAgent.CapabilitiesWithOutSecret | ConvertTo-Json
+                }
+                else 
+                {
+                    $body =$CurrentAgent.CapabilitiesWithSecret | ConvertTo-Json
+    
+                }  
+                $url = "https://dev.azure.com/{0}/_apis/distributedtask/pools/{1}/agents/{2}/usercapabilities?api-version=5.0-preview.1" -f $($this.OrganizationContext.OrganizationName),$($CurrentAgent.PoolId), $($CurrentAgent.AgentId);          
```

Comment:
$this.OrganizationContext.OrganizationName , $CurrentAgent.PoolId, $CurrentAgent.AgentId; is enough, $($ not needed